pkg/apis/pipeline/v1beta1/taskrun_types.go

```diff
@@ -66,6 +66,10 @@ type TaskRunSpec struct {
 	// Workspaces is a list of WorkspaceBindings from volumes to workspaces.
 	// +optional
 	Workspaces []WorkspaceBinding `json:"workspaces,omitempty"`
+	// ExecutionMode specifies the execution mode to run user-specified containers in
+	// (tekton injected containers will not run with this policy)
+	// +optional
+	ExecutionMode ExecutionMode `json:"executionMode,omitempty"`
```

Comment:
We had discussed including this as an annotation (or maybe only in the v1alpha1 API only)? first, to indicate that it's experimental: https://github.com/tektoncd/community/blob/main/teps/0025-hermekton.md#implementation-plan

I'm not quite sure what the best approach would be here. The API docs don't really say either: https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md
Probably a good topic for the API working group.

Reply:
We now have an feature gated API fields with related testing helpers that can be used to introduce "alpha" level features into the beta API: https://github.com/tektoncd/pipeline/tree/main/docs/developers#adding-feature-gated-api-fields

Reply:
Agreed on discussing this in the API working group. We should try to not add a new field at first (I tend to like the annotation approach) ; and if we do, we should *hide* it behind the alpha api feature flag (cc @sbwsg )

Reply:
+1, so something like:

* annotation behind alpha feature flag
* beta field behind alpha feature flag
* beta field without feature flag ?

Reply:
sgtm, I switched to an annotation and put it behind the alpha feature flag